pkg/clients/gitlab/git.go

```diff
@@ -120,3 +121,29 @@ func (gc *GitlabClient) CloseMergeRequest(projectID string, mergeRequestIID int)
 
 	return nil
 }
+
+// DeleteWebhooks deletes webhooks in Gitlab repo by given project ID
+func (gc *GitlabClient) DeleteWebhooks(projectID, clusterAppDomain string) error {
+
+	// Check if clusterAppDomain is empty returns error, else continue
+	if clusterAppDomain == "" {
+		return fmt.Errorf("ClusterAppDomain is empty: %s", clusterAppDomain)
+	}
+
+	// List project hooks
+	webhooks, _, err := gc.client.Projects.ListProjectHooks(projectID, nil)
+	if err != nil {
+		return fmt.Errorf("failed to list project hooks: %v", err)
+	}
+
+	// Delete matching webhooks
+	for _, webhook := range webhooks {
+		if strings.Contains(webhook.URL, clusterAppDomain) {
+			if _, err := gc.client.Projects.DeleteProjectHook(projectID, webhook.ID); err != nil {
+				return fmt.Errorf("failed to delete webhook (ID: %d): %v", webhook.ID, err)
+			}
+		}
```

Comment:
In case we expect that number of webhooks, is the for cycle needed then?

Reply:
good point @dheerajodha ,  
it would better practice I add the break once webhook is found , 
@jsztuka, we expect that one webhook per each  ConsoleAppURL (konflux cluster)